backend/msgpack/conn.go

```diff
@@ -165,8 +171,12 @@ func (c *connection) writeWithLock(data []byte) error {
 	if err := c.conn.SetWriteDeadline(c.nowFn().Add(c.writeTimeout)); err != nil {
 		c.metrics.setWriteDeadlineError.Inc(1)
 	}
-	_, err := c.conn.Write(data)
-	return err
+	if _, err := c.conn.Write(data); err != nil {
+		c.metrics.writeError.Inc(1)
+		return err
+	}
+	c.lastEvent = c.nowFn()
```

Comment:
Do we need to record the time when writes succeed? I'd think if you have a string of successful writes, and the first time your write fails, we'll always try to reconnect at line 113, which updates the time of last connection attempt, so feels cleaner to only track `lastConnectNanos`. Wdyt?

Reply:
Yea, good point, it's unnecessary, will remove.